index.js

```diff
@@ -6,43 +6,47 @@ const DEFAULT_APP_NAME = 'node';
 
 /**
  * @param {object} opts An object containing the required configuration.
- *    app                         - An express applcation
- *    appName                     - The name to be used as a default label
- *    ignoredRoutes (optional)    - An array of routes that should be ignored
+ *    appName                           - The name to be used as a default label
+ *    ignoredRoutes (optional)          - An array of routes that should be ignored
+ *    collectDefaultMetrics (optional)  - A boolean indicating whether or not to collect default metrics (default: true)
  * @returns {object} The internal prometheus client, middleware function, and metrics function.
  */
-module.exports = ({appName = DEFAULT_APP_NAME, ignoredRoutes = DEFAULT_IGNORED_ROUTES}) => {
+module.exports = ({appName = DEFAULT_APP_NAME, ignoredRoutes = DEFAULT_IGNORED_ROUTES, collectDefaultMetrics = false}) => {
 	Prometheus.register.setDefaultLabels({appName});
-	Prometheus.collectDefaultMetrics({timeout: DEFAULT_TIMEOUT_MS});
 
-	const durrationSummary = createDurrationSummary();
+	const durationSummary = createDurationSummary();
 	const statusCounter = createStatusCounter();
+    let defaultMetricsInterval = null;
 
-	/**
-   * This is the middleware function that actually measures response metrics. For every request the
-   * durration is measured as well as the request path, request method, and response status.
-   * @param {Request} req A request object
-   * @param {Response} res A responsee object
-   * @param {Function} next Function to triggre the next middleware function in the chain
-   */
+    if (collectDefaultMetrics) {
+        defaultMetricsInterval = Prometheus.collectDefaultMetrics({timeout: DEFAULT_TIMEOUT_MS});
+    }
+
+    /**
+     * This is the middleware function that actually measures response metrics. For every request the
+     * duration is measured as well as the request path, request method, and response status.
+     * @param {Request} req A request object
+     * @param {Response} res A response object
+     * @param {Function} next Function to trigger the next middleware function in the chain
+     */
 	const middlewareFunc = (req, res, next) => {
 		if (!ignoredRoutes.includes(req.path)) {
-			const end = durrationSummary.startTimer();
+			const end = durationSummary.startTimer();
 			res.on('finish', () => {
-				const reqData = {method: req.method, path: req.path, status: res.statusCode};
+				const reqData = {status: res.statusCode};
```

Comment:
Ah - thats a good solution actually. That being said, we're not going to be monitoring specific routes - only overall metrics. So while this would allow us to reduce the unique URLS, we'd still be collecting a bunch of data that we don't actually need (yet).